netket/machine/jax.py

```diff
@@ -332,3 +332,95 @@ def JaxRbm(hilbert, alpha, dtype=complex):
         stax.serial(stax.Dense(alpha * hilbert.size), LogCoshLayer, SumLayer()),
         dtype=dtype,
     )
+
+
+def JaxMpsPeriodic(hilbert, bond_dim, diag=False, symperiod=-1, dtype=complex):
+    return Jax(hilbert, stax.serial(MpsPeriodicLayer(hilbert, bond_dim, diag, symperiod, dtype)), dtype=dtype)
+
+
+def MpsPeriodicLayer(hilbert, bond_dim, diag=False, symperiod=-1, dtype=complex):
+    #default standard deviation equals 1e-2
+    normal_init = jax.nn.initializers.normal()
+
+    L = hilbert.size
+    phys_dim = hilbert.local_size
+    diag = diag
+    dtype = dtype
+
+    #determine transformation from local states to indices
+    local_states = jax.numpy.array(hilbert.local_states)
+    loc_vals_spacing = jax.numpy.roll(local_states, -1)[0:-1] - local_states[0:-1]
+    if jax.numpy.max(loc_vals_spacing) == jax.numpy.min(loc_vals_spacing):
+        loc_vals_spacing = loc_vals_spacing[0]
+    else:
+        raise AssertionError("JaxMpsPeriodic can only be used with evenly spaced hilbert local values")
+    loc_vals_bias = jax.numpy.min(local_states)
+
+    #determine shape of unit cell
+    if symperiod < 0:
```

Comment:
It would be nice to check if the system is 1d and the symmetrization here is properly used, maybe one could check if the lattice is isomorphic to a 1d chain? (we have networkx included by default now)

Reply:
We can certainly check that the lattice is isomorphic to a 1d chain.  However one might want to leave the freedom of using an mps also on different lattices to the user?  For example I can imagine people wanting to try snaking an mps on a 2D lattice.  Maybe we can check the lattice and print a warning when it is not isomorphic to a 1D chain.  

I don't know how one could check the symmetrization because this depends also on the Hamiltonian (e.g. unit cell with period 2 in AFH and unit cell with period 1 in TFI).  So the correct symmetry depends on the physics which is hard to check for generic Hamiltonians.

Reply:
You are right, maybe this is a bit too much. It's fine to have it unchecked I guess... !